src/modules/files-storage/domain/service/files-storage.service.ts

```diff
@@ -146,40 +164,47 @@ export class FilesStorageService {
 		return fileName;
 	}
 
-	private async createFileInStorageAndRollbackOnError(fileRecord: FileRecord, file: FileDto): Promise<void> {
-		const filePath = fileRecord.createPath();
+	private async createFileInStorageAndDeleteOnError(fileRecord: FileRecord, file: FileDto): Promise<void> {
+		try {
+			await this.storeAndScanFile(file, fileRecord);
+		} catch (error) {
+			const filePath = fileRecord.createPath();
+
+			await this.storageClient.delete([filePath]);
+			await this.fileRecordRepo.delete(fileRecord);
+
+			throw error;
+		}
+	}
+
+	private async storeAndScanFile(file: FileDto, fileRecord: FileRecord): Promise<void> {
 		const useStreamToAntivirus = this.config.FILES_STORAGE_USE_STREAM_TO_ANTIVIRUS;
+		const fileSizePromise = this.countFileSize(file);
+		const filePath = fileRecord.createPath();
 
-		try {
-			const fileSizePromise = this.countFileSize(file);
-
-			if (useStreamToAntivirus && fileRecord.isPreviewPossible()) {
-				const streamToAntivirus = file.data.pipe(new PassThrough());
-
-				const [, antivirusClientResponse] = await Promise.all([
-					this.storageClient.create(filePath, file),
-					this.antivirusService.checkStream(streamToAntivirus),
-				]);
-				const { status, reason } = ScanResultDtoMapper.fromScanResult(antivirusClientResponse);
-				fileRecord.updateSecurityCheckStatus(status, reason);
-			} else {
-				await this.storageClient.create(filePath, file);
-			}
+		if (useStreamToAntivirus && fileRecord.isPreviewPossible()) {
+			const streamToAntivirus = file.data.pipe(new PassThrough());
+
+			const [, antivirusClientResponse] = await Promise.all([
+				this.storageClient.create(filePath, file),
+				this.antivirusService.checkStream(streamToAntivirus),
+			]);
+			const { status, reason } = ScanResultDtoMapper.fromScanResult(antivirusClientResponse);
+			fileRecord.updateSecurityCheckStatus(status, reason);
+		} else {
+			await this.storageClient.create(filePath, file);
+		}
 
-			// The actual file size is set here because it is known only after the whole file is streamed.
-			const fileRecordSize = await fileSizePromise;
+		// The actual file size is set here because it is known only after the whole file is streamed.
+		const fileRecordSize = await fileSizePromise;
```

Comment:
const fileSizePromise = this.countFileSize(file);  --> const fileSizeObserver = StreamFileSizeObserverFactory(file: FileDto)

const fileRecordSize = await fileSizePromise; --> const fileRecordSize = await fileSizeObserver.calculateFileSize();

Kommentar kann dann entfernt werden.